tracer/src/Datadog.Trace/AppSec/Rasp/MetaStructHelper.cs

```diff
@@ -0,0 +1,95 @@
+// <copyright file="MetaStructHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Vendors.MessagePack.Formatters;
+
+#nullable enable
+
+namespace Datadog.Trace.AppSec.Rasp;
+
+internal static class MetaStructHelper
+{
+    public static Dictionary<string, object> StackTraceInfoToDictionary(string? type, string language, string id, string? message, List<Dictionary<string, object>> frames)
+    {
+        var dict = new Dictionary<string, object>(3)
+        {
+            { "language", language },
+            { "id", id },
+            { "frames", frames }
+        };
+
+        if (type is not null && type.Length > 0)
+        {
+            dict["type"] = type;
+        }
+
+        if (message is not null && message.Length > 0)
+        {
+            dict["message"] = message;
+        }
+
+        return dict;
+    }
+
+    public static Dictionary<string, object> StackFrameToDictionary(uint id, string? text, string? file, uint? line, uint? column, string? ns, string? className, string? function)
+    {
+        var dict = new Dictionary<string, object>(7)
```

Comment:
You add 8 items I think?
```suggestion
        var dict = new Dictionary<string, object>(8)
```

Reply:
I tend to think it's probably better to allocate more items that necessary, to avoid the need to grow the dict if becomes full. However, the advantages are debatable, so I'd probably say leave it as it is for now.